### PR TITLE
Step Bundles, Iteration 2: Add Step Bundles entity page

### PR DESCRIPTION
**Pull request by k-olly**

### CODE REVIEW

File: source/javascripts/pages/StepBundlesPage/StepBundlesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }));
+  const hasStepBundles = stepBundlesIds.length > 0;
+
+  useEffect(() => {
+    closeDialog();
+  }, [hasStepBundles, closeDialog]);
+
+  if (!hasStepBundles) {
+    return (
+      <Box h="100%" display="grid" gridTemplateRows="100%">
```
Comment (moczolaszlo) [inline source/javascripts/pages/StepBundlesPage/StepBundlesPage.tsx:27]: I think this Box isnt needed here.

File: source/javascripts/components/unified-editor/StepSelectorDrawer/components/SelectableStepBundleCard.tsx [hunk 2]
```diff
@@ -31,6 +31,7 @@ const SelectableStepBundleCard = (props: SelectableStepBundleCardProps) => {
           textAlign="left"
           _hover={{ borderColor: 'border/hover' }}
           onClick={handleClick}
+          marginBottom="12"
```
Comment (moczolaszlo) [inline source/javascripts/components/unified-editor/StepSelectorDrawer/components/SelectableStepBundleCard.tsx:34]: Please avoid this inside a component. Use flex for the parent list, or add margin from a prop.
I know it works here, because we use this component only one place. But its a bad pattern - if we need this component somewhere else this margin will be disturbing.

File: source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { arrayMove, SortableContext, verticalListSortingStrategy } from '@dnd-ki
 
 import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
 
-import { useStepActions } from '../contexts/WorkflowCardContext';
+import { useStepActions } from '@/components/unified-editor/WorkflowCard/contexts/WorkflowCardContext';
```
Comment (moczolaszlo) [inline source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx:10]: 🙅

**Review by moczolaszlo — CHANGES_REQUESTED** [review]

File: source/javascripts/components/unified-editor/StepBundleDrawer/StepBundlePropertiesTab.tsx [hunk 4]
```diff
@@ -1,31 +1,56 @@
+import { Button, useDisclosure } from '@bitrise/bitkit';
 import EditableInput from '@/components/EditableInput/EditableInput';
 import StepBundleService from '@/core/models/StepBundleService';
 import { useStepBundles } from '@/hooks/useStepBundles';
-import useRenameStepBundle from './hooks/useRenameStepBundle';
+import DeleteStepBundleDialog from '@/pages/StepBundlesPage/components/DeleteStepBundleDialog/DeleteStepBundleDialog';
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/unified-editor/StepBundleDrawer/StepBundlePropertiesTab.tsx:5]: Don't reference components under from a page, move the component to here instead

File: source/javascripts/components/unified-editor/StepSelectorDrawer/components/SelectableStepBundleCard.tsx [hunk 5]
```diff
@@ -42,7 +42,13 @@ const SelectableStepBundleCard = (props: SelectableStepBundleCardProps) => {
       </PopoverTrigger>
       <PopoverContent width={320}>
         <WorkflowCardContextProvider>
-          <StepBundleCard uniqueId="" stepIndex={-1} cvs={`bundle::${id}`} isPreviewMode={isPreviewMode} />
+          <StepBundleCard
+            uniqueId=""
+            stepIndex={-1}
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/unified-editor/StepSelectorDrawer/components/SelectableStepBundleCard.tsx:47]: Why does it have a `stepIndex` prop?

File: source/javascripts/pages/StepBundlesPage/components/Drawers.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+          workflowId=""
+          stepIndex={stepIndex}
+          isOpen={isDialogOpen(StepBundlesPageDialogType.STEP_CONFIG)}
+          onClose={closeDialog}
+          onCloseComplete={unmountDialog}
+        />
+      )}
+
+      {isDialogMounted(StepBundlesPageDialogType.STEP_BUNDLE) && (
+        <StepBundleDrawer
```
Comment (AndrasEszes) [inline source/javascripts/pages/StepBundlesPage/components/Drawers.tsx:64]:
```suggestion
        <StepBundleConfigDrawer
```

File: source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx [hunk 7]
```diff
@@ -88,10 +87,17 @@ const StepBundleStepList = ({ stepBundleId, isPreviewMode, ...actions }: Props)
                 <AddStepButton
                   onClick={() => onAddStepToStepBundle(stepBundleId, item.stepIndex)}
                   showStepBundles={false}
+                  my={isPreviewMode ? '0px' : '-8px'}
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx:90]: In preview mode, does it even show up? (it shouldn't)

File: source/javascripts/pages/StepBundlesPage/components/StepBundlesConfigPanel/StepBundlesConfig.context.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+const Context = createContext<State>(undefined);
+
+type Props = PropsWithChildren<{ stepBundleId: string }>;
+const StepBundlesConfigProvider = ({ stepBundleId, children }: Props) => {
+  const stepBundle = useStepBundle(stepBundleId);
+
+  return <Context.Provider value={stepBundle as StepBundle}>{children}</Context.Provider>;
+};
+
+export const useWorkflowConfigContext = () => {
```
Comment (AndrasEszes) [inline source/javascripts/pages/StepBundlesPage/components/StepBundlesConfigPanel/StepBundlesConfig.context.tsx:15]:
```suggestion
export const useStepBundleConfigContext = () => {
```

File: source/javascripts/core/models/BitriseYmlService.spec.ts [hunk 9]
```diff
@@ -3218,6 +3218,69 @@ describe('BitriseYmlService', () => {
...
+      };
+
+      const actualYml = BitriseYmlService.createStepBundle('bundle2', sourceYml, 'bundle1');
+
+      expect(actualYml).toMatchBitriseYml(expectedYml);
+    });
+  });
+
+  describe('deleteStepBundle', () => {
+    it('should remove a step bundle in the whole yml completely', () => {
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/core/models/BitriseYmlService.spec.ts:3260]: Add some additional edge case tests

File: source/javascripts/pages/StepBundlesPage/components/StepBundlesConfigPanel/StepBundlesConfigPanel.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+  const handleOnDelete = () => {
+    setStepBundleId('');
+    closeDialog();
+  };
+
+  return (
+    <Box borderLeft="1px solid" borderColor="border/regular">
+      <StepBundlesConfigHeader parentStepBundleId={stepBundleId} />
+      <Box padding="16px 24px">
```
Comment (AndrasEszes) [inline source/javascripts/pages/StepBundlesPage/components/StepBundlesConfigPanel/StepBundlesConfigPanel.tsx:22]: Move into the StepBundlePropertiesTab

File: source/stylesheets/_index.scss.erb
Comment (AndrasEszes) [inline source/stylesheets/_index.scss.erb:1]: 😬

File: source/javascripts/pages/StepBundlesPage/components/CreateStepBundleDialog/CreateStepBundleDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import { DialogProps } from '@bitrise/bitkit';
+import CreateEntityDialog from '@/components/unified-editor/CreateEntityDialog/CreateEntityDialog';
+import StepBundleService from '@/core/models/StepBundleService';
+import { useStepBundles } from '@/hooks/useStepBundles';
+
+type Props = Omit<DialogProps, 'title'> & {
+  onCreateStepBundle: (stepBundleId: string, baseStepBundleId?: string) => void;
+};
+
+const CreateStepBundleDialog = ({ onClose, onCloseComplete, onCreateStepBundle, ...props }: Props) => {
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/pages/StepBundlesPage/components/CreateStepBundleDialog/CreateStepBundleDialog.tsx:10]: This can be moved under the unified-editor in components

File: source/javascripts/pages/StepBundlesPage/components/DeleteStepBundleDialog/DeleteStepBundleDialog.tsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { Button, Dialog, DialogBody, DialogFooter, DialogProps, List, ListItem, Text } from '@bitrise/bitkit';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
+
+type Props = Omit<DialogProps, 'title'> & {
+  stepBundleId: string;
+  onDeleteStepBundle?: (stepBundleId: string) => void;
+};
+
+const DeleteStepBundleDialog = ({ stepBundleId, onDeleteStepBundle, onClose, ...dialogProps }: Props) => {
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/pages/StepBundlesPage/components/DeleteStepBundleDialog/DeleteStepBundleDialog.tsx:10]: Move under unified-editor

File: source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx (hunk 7 above)
Comment (k-olly) [reply source/javascripts/components/unified-editor/WorkflowCard/components/StepBundleStepList.tsx:90]: Nice catch, thanks

File: source/javascripts/pages/StepBundlesPage/components/DeleteStepBundleDialog/DeleteStepBundleDialog.tsx (hunk 12 above)
Comment (zoltanszabo-bitrise) [reply source/javascripts/pages/StepBundlesPage/components/DeleteStepBundleDialog/DeleteStepBundleDialog.tsx:10]: Don't we have a generic DeleteEntityDialog yet?

**Review by AndrasEszes — APPROVED** [review]
